cmd/kafka-example/main.go

```diff
@@ -56,41 +57,40 @@ func main() {
 		close(ch)
 	}(stdinChannel)
 
-	go func() {
-		for {
-			select {
-			case consumedMessage := <-consumer.Incoming():
-				consumedData := consumedMessage.GetData()
-				log.Info("Message consumed", log.Data{
-					"messageString": string(consumedData),
-					"messageRaw":    consumedData,
-					"messageLen":    len(consumedData),
-				})
-				consumedMessage.Commit()
-			case consumerError := <-consumer.Errors():
-				log.Error(fmt.Errorf("Aborting"), log.Data{"messageReceived": consumerError})
-				producer.Closer() <- true
-				consumer.Closer() <- true
-				exitChannel <- true
-				return
-			case producerError := <-producer.Errors():
-				log.Error(fmt.Errorf("Aborting"), log.Data{"messageReceived": producerError})
-				producer.Closer() <- true
-				consumer.Closer() <- true
-				exitChannel <- true
-				return
-			case stdinLine := <-stdinChannel:
-				producer.Output() <- []byte(stdinLine)
-				log.Info("Message output", log.Data{"messageSent": stdinLine})
-			case <-signals:
-				log.Info("Quitting after signal", nil)
-				producer.Closer() <- true
-				consumer.Closer() <- true
-				exitChannel <- true
-				return
-			}
+	shutdownGracefully := func() {
+
+		// give the app 3 seconds to close gracefully before killing it.
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Minor comment - but do we want to make the timeout configurable?

Reply:
Probably not for the example, but definitely in production apps